File: fixtures/lua/src/calculator.lua
local operations = require("src.operations")

local Calculator = {}
Calculator.__index = Calculator

--- Create a new Calculator instance.
--- @return table
function Calculator.new()
    local self = setmetatable({}, Calculator)
    self._history = {}
    return self
end

--- Add two numbers and record the result.
--- @param a number
--- @param b number
--- @return number
function Calculator:add(a, b)
    local result = operations.add(a, b)
    table.insert(self._history, result)
    return result
end

--- Subtract b from a and record the result.
--- @param a number
--- @param b number
--- @return number
function Calculator:subtract(a, b)
    local result = operations.subtract(a, b)
    table.insert(self._history, result)
    return result
end

--- Multiply two numbers and record the result.
--- @param a number
--- @param b number
--- @return number
function Calculator:multiply(a, b)
    local result = operations.multiply(a, b)
    table.insert(self._history, result)
    return result
end

--- Divide a by b and record the result.
--- @param a number
--- @param b number
--- @return number
function Calculator:divide(a, b)
    local result = operations.divide(a, b)
    table.insert(self._history, result)
    return result
end

--- Return a copy of the calculation history.
--- @return table
function Calculator:get_history()
    local copy = {}
    for i, v in ipairs(self._history) do
        copy[i] = v
    end
    return copy
end

--- Clear the calculation history.
function Calculator:clear_history()
    self._history = {}
end

return Calculator

File: fixtures/javascript/src/calculator.js
import { add, subtract, multiply, divide } from './operations.js';

export class Calculator {
  #history = [];

  add(a, b) {
    const result = add(a, b);
    this.#history.push(result);
    return result;
  }

  subtract(a, b) {
    const result = subtract(a, b);
    this.#history.push(result);
    return result;
  }

  multiply(a, b) {
    const result = multiply(a, b);
    this.#history.push(result);
    return result;
  }

  divide(a, b) {
    const result = divide(a, b);
    this.#history.push(result);
    return result;
  }

  getHistory() {
    return [...this.#history];
  }

  clearHistory() {
    this.#history = [];
  }
}
